### PR TITLE
Safe entity copying

### PR DESCRIPTION
**Pull request by ddavidov**
- Check alias / name on copying and generate safe alias before copy save

### CODE REVIEW

File: zoolanders/Framework/Listener/Item/BeforeCopy.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+use Zoolanders\Framework\Listener\Listener;
+use Zoolanders\Framework\Event\Item\Copy;
+use Zoolanders\Framework\Model\Item as ItemModel;
+
+/**
+ * Class BeforeCopy
+ * @package Zoolanders\Framework\Listener\Item
+ */
+class BeforeCopy extends Listener
```
Comment (Skullbock) [inline zoolanders/Framework/Listener/Item/BeforeCopy.php]: The listener class name should reference the action taken, not the event listening. So something like CreateAlias

File: zoolanders/Framework/Event/Item/Copy.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * @package     ZOOlanders Framework
+ * @version     4.0.0-beta11
+ * @author      ZOOlanders - http://zoolanders.com
+ * @license     GNU General Public License v2 or later
+ */
+
+namespace Zoolanders\Framework\Event\Item;
+
+class Copy extends Item
```
Comment (Skullbock) [inline zoolanders/Framework/Event/Item/Copy.php]: This one should be BeforeCopy

File: zoolanders/Framework/Model/Database/UniqueAlias.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     * Builds safe alias for provided one
+     *
+     * @param   $alias
+     *
+     * @return  array alias, copy_number
+     */
+    public function makeSafeAlias($alias){
```
Comment (Skullbock) [inline zoolanders/Framework/Model/Database/UniqueAlias.php]: let's use generateAlias

File: zoolanders/Framework/Listener/Item/BeforeCopy.php (hunk 1 above)
Comment (ddavidov) [inline zoolanders/Framework/Listener/Item/BeforeCopy.php]: Ok, got it

File: zoolanders/Framework/Event/Item/Copy.php (hunk 2 above)
Comment (ddavidov) [inline zoolanders/Framework/Event/Item/Copy.php]: Ok

File: zoolanders/Framework/Model/Database/UniqueAlias.php (hunk 3 above)
Comment (ddavidov) [inline zoolanders/Framework/Model/Database/UniqueAlias.php]: Ok

**Pull request merged**